src/main/java/PersistentStorageHandler.java

```diff
@@ -0,0 +1,61 @@
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+
+class PersistentStorageHandler {
+
+    private static final String TASKLIST_PATH = "./tasklist.dat";
+
+    public PersistentStorageHandler() { }
+
+    public boolean ensureTaskFileExists() throws DukeException {
+        File file = new File(TASKLIST_PATH);
+
+        if (!file.exists()) {
+            try {
+                file.createNewFile();
+                return false;
+            } catch (IOException e) {
+                System.out.println(e.getMessage());
+                throw new DukeException("Failed to create persistent task file.");
+            }
+        }
+        return true;
+    }
+
+    private TaskList decode(ObjectInputStream ois) throws DukeException {
```

Comment:
Perhaps this method name could be more descriptive? Such as decodeFileContents()?